simbad/mr/anomalous_util.py

```diff
@@ -132,63 +123,22 @@ def run(self, model):
-        pdb_input = iotbx.pdb.pdb_input(file_name=heavy_atom_model)
-        hierarchy = pdb_input.construct_hierarchy()
-        for residue_group in hierarchy.models()[0].chains()[0].residue_groups():
-            for atom_group in residue_group.atom_groups():
-                for atom in atom_group.atoms():
-                    if atom.b >= 6:
-                        peaks_over_6_rms_coordinates.append((atom.xyz[0], atom.xyz[1], atom.xyz[2]))
-                    if atom.b >= 9:
-                        peaks_over_9_rms_coordinates.append((atom.xyz[0], atom.xyz[1], atom.xyz[2]))
-
-        # Get the atom coordinates of the placed MTZ
-        input_model_coordinates = []
-
-        pdb_input = iotbx.pdb.pdb_input(file_name=input_model)
-        hierarchy = pdb_input.construct_hierarchy()
-
-        for residue_group in hierarchy.models()[0].chains()[0].residue_groups():
-            for atom_group in residue_group.atom_groups():
-                for atom in atom_group.atoms():
-                    input_model_coordinates.append((atom.xyz[0], atom.xyz[1], atom.xyz[2]))
-
-        # Find the number of peaks within min dist (default 2A) of protein
-        peaks_over_6_rms_within_4 = 0
-        peaks_over_9_rms_within_4 = 0
-
-        # Find the number of peaks over 8 rms that have an euclidean distance from the protein of less than min_dist
-        for peak_coordinate in peaks_over_6_rms_coordinates:
-            for atom_coordinate in input_model_coordinates:
-                dist = distance.euclidean(peak_coordinate, atom_coordinate)
-                if dist <= min_dist:
-                    peaks_over_6_rms_within_4 += 1
-                    break
-
-        # Find the number of peaks over 12 rms that have an euclidean distance from the protein of less than min_dist
-        for peak_coordinate in peaks_over_9_rms_coordinates:
-            for atom_coordinate in input_model_coordinates:
-                dist = distance.euclidean(peak_coordinate, atom_coordinate)
-                if dist <= min_dist:
-                    peaks_over_9_rms_within_4 += 1
-                    break
-
-        score = _AnomScore(peaks_over_6_rms=len(peaks_over_6_rms_coordinates),
-                           peaks_over_9_rms=len(peaks_over_9_rms_coordinates),
-                           peaks_over_6_rms_within_4a_of_model=peaks_over_6_rms_within_4,
-                           peaks_over_9_rms_within_4a_of_model=peaks_over_9_rms_within_4)
+        heavy_atom_file = os.path.join(self.work_dir, "peakmax_{0}.ha".format(self.name))
+        all_peaks = []
+
+        with open(heavy_atom_file, 'r') as f:
+            for line in f:
+                if line.startswith('ATOM'):
+                    peak = line.split()[5]
+                    all_peaks.append(peak)
+
+        z_scores = stats.zscore(numpy.array(all_peaks))
+
+        score = _AnomScore(dano_peak_height=all_peaks[0],
+                           dano_z_score=float(max(z_scores)))
```

Comment:
Could just use ``stats.zscore(all_peaks).max()``, which should return a `float`.